src/main/java/net/snowflake/client/jdbc/SnowflakeConnectionV1.java

```diff
@@ -1060,7 +1013,7 @@ public void setInjectFileUploadFailure(String fileToFail) throws SQLException {
     sfSession.setInjectFileUploadFailure(fileToFail);
   }
 
-  public SFSession getSfSession() {
+  public SFBaseSession getSfSession() {
```

Comment:
The user may call `getSfSession` to also retrieve other implementations of Session. One example of this is in Snowpark, where the code will call `getSfSession`, and we need to return the StoredProcSession. So we need to return the base-class type.

Reply:
What about have both `getSFSession` and `getSFBaseSession` APIs?

Reply:
Hmm, what if the current `Connection` is using a different Session type? What should `getSfSession()` return? `null`?

Each `SnowflakeConnection` should only have one `SFBaseSession` object active. You should never have `SFSession` in addition to another `SFBaseSession`. If we are using `SFSession`, then `SFSession` will be the active `SFBaseSession`.

Reply:
so the getSFSession can be somthine like
```
SFSession getSFSession() {
  if (session isinstanceOf SFSession) {
    return (SFSession) session
  }
  throw SomeException;
}
```

Reply:
Sure, we can do that. Just curious about the reason for this: is the purpose for convenience, since you expect 99% of JDBC use to still use `SFSession`?

Reply:
My rationale is (not necessary always right):
1. yes for convenience, you don't need to change other code; also the name makes sense to get a SFSession from getSFSession()
2. is more for coding style. We provide an API to directly get SFSession and cover the casting and exception in one place so the consumers of SnowflakeConnectionV1 don't need to do it each time they try to access it (I believe this API will be used a lot)

Reply:
Sounds good.